sdk/feature-management/test/featureEvaluation.test.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as chai from "chai";
+const expect = chai.expect;
+
+import { FeatureManager, ConfigurationObjectFeatureFlagProvider, EvaluationResult, VariantAssignmentReason } from "../";
+
+let called: boolean = false;
+const dummyCallback = () => {
+    called = true;
+};
+
+let evaluationResult: EvaluationResult;
+const setEvaluationResult = (result: EvaluationResult) => {
+    evaluationResult = result;
+};
+
+describe("feature evaluation", () => {
+    it("should not call onFeatureEvaluated when telemetry is not enabled", async () => {
+        const jsonObject = {
+            "feature_management": {
+                "feature_flags": [
+                    {
+                        "id": "TestFeature",
+                        "enabled": true,
+                        "telemetry": { "enabled": false }
+                    }
+                ]
+            }
+        };
+
+        const provider = new ConfigurationObjectFeatureFlagProvider(jsonObject);
+        const featureManager = new FeatureManager(provider, { onFeatureEvaluated: dummyCallback});
+
+        await featureManager.isEnabled("TestFeature");
+        expect(called).to.eq(false);
+    });
+
+    it("should not assign variant when there is no variants", async () => {
+        const jsonObject = {
+            "feature_management": {
+                "feature_flags": [
+                    {
+                        "id": "TestFeature",
+                        "enabled": true,
+                        "allocation": { "default_when_enabled": "Big" },
+                        "telemetry": { "enabled": true}
+                    }
+                ]
+            }
+        };
+
+        const provider = new ConfigurationObjectFeatureFlagProvider(jsonObject);
+        const featureManager = new FeatureManager(provider, { onFeatureEvaluated: setEvaluationResult});
+
+        await featureManager.isEnabled("TestFeature");
+        expect(evaluationResult.feature?.id).to.eq("TestFeature");
+        expect(evaluationResult.enabled).to.eq(true);
+        expect(evaluationResult.userId).to.eq(undefined);
+        expect(evaluationResult.variant).to.eq(undefined);
+        expect(evaluationResult.variantAssignmentReason).to.eq(VariantAssignmentReason.None);
+
+        evaluationResult = new EvaluationResult(undefined);
```

Comment:
I think you should pull all of these out into a beforeEach

```javascript
beforeEach(function() {
  evaluationResult = new EvaluationResult(undefined);
}
```

Also I would expect you to clear it rather than construct an undefined one.

```javascript
beforeEach(function() {
  evaluationResult = null;
}
```